apps/web/src/components/Spaces/Common/AvatarGrid/Avatar.tsx

```diff
@@ -0,0 +1,119 @@
+import { useEventListener, useHuddle01 } from '@huddle01/react/hooks';
+import { Image } from '@lenster/ui';
+import type { FC } from 'react';
+import React from 'react';
+import { useSpacesStore } from 'src/store/spaces';
+import { useUpdateEffect } from 'usehooks-ts';
+
+import { NestedPeerListIcons } from '../assets/Icons';
+import Audio from '../Audio';
+import Dropdown from '../Dropdown';
+import CoHostData from '../Sidebar/Peers/PeerRole/CoHostData';
+import HostData from '../Sidebar/Peers/PeerRole/HostData';
+import ListenersData from '../Sidebar/Peers/PeerRole/ListenersData';
+import SpeakerData from '../Sidebar/Peers/PeerRole/SpeakerData';
+import type { IRoleEnum } from '../SpacesTypes';
+
+type Props = {
+  peerId: string;
+  displayName: string;
+  mic?: MediaStreamTrack | null;
+  role: IRoleEnum;
+  avatarUrl: string;
+};
+
+const Avatar: FC<Props> = ({ peerId, displayName, mic, role, avatarUrl }) => {
+  const { isMyHandRaised, setMyHandRaised, myReaction, setMyReaction } =
+    useSpacesStore();
+
+  const { me } = useHuddle01();
+
+  const RoleData = {
+    host: <HostData />,
+    coHost: <CoHostData peerId={peerId} />,
+    speaker: <SpeakerData peerId={peerId} />,
+    listener: <ListenersData peerId={peerId} />
+  } as const;
+
+  useEventListener('room:data-received', (data) => {
+    if (data.fromPeerId === peerId && data.payload['reaction']) {
+      setMyReaction(data.payload['reaction']);
+      setTimeout(() => {
+        setMyReaction('');
+      }, 5000);
+    }
+    if (
+      data.fromPeerId === peerId &&
+      (data.payload['raiseHand'] == true || data.payload['raiseHand'] == false)
```

Comment:
what kind of condition is this and what other result are we anticipating here?

Reply:
Here, I want that data.payload['raiseHand'] should have true or false. This is used to read whether the hand is raised or not.

Reply:
do we need this condition then?